capella_ros_tools/exporter.py

```diff
@@ -2,83 +2,625 @@
+                card = ("1", "1")
+
+            if card != ("1", "1"):
+                if card[1] == "*":
+                    type_name += "[]"
+                elif card[0] == card[1]:
+                    type_name += f"[{card[0]}]"
+                else:
+                    type_name += f"[<={card[1]}]"
+
+            cls_data.literals.append(
+                LiteralData(
+                    type_name,
+                    prop_name,
+                    docstr=self._make_doc_str(prop.description),
+                )
+            )
+
+        return cls_data
+
+    def _handle_property_type(
+        self,
+        _type: capellambse.model.ModelElement,
+        class_package_mapping: dict[str, str],
+        cls: information.Class,
+        cls_data: ClassData,
+        current_pkg: str,
+        pkg_cls_uuids: typing.Iterable[str],
+        pkg_dependencies: set[str],
+        prop_name: str,
+    ) -> str:
+        if isinstance(_type, information.datatype.DataType):
+            if isinstance(_type, information.datatype.Enumeration):
+                if _type.domain_type:
+                    type_name = self._make_type_name(_type.domain_type)
+                else:
+                    logger.warning(
+                        "Primitive type of %s should be added as "
+                        "domain_type, will use int32 instead",
+                        _type.name,
+                    )
+                    type_name = "int32"
+                for val in _type.owned_literals:
+                    cls_data.literals.append(
+                        LiteralData(
+                            type_name,
+                            f"{prop_name.upper()}_{self.make_snake_case(val.name).upper()}",
+                            val.value.value,
+                            self._make_doc_str(val.description),
+                        )
+                    )
+            else:
+                type_name = self._make_type_name(_type)
+        elif isinstance(_type, information.Class):
+            pkg = ""
+            build_in = False
+            if _type.uuid not in pkg_cls_uuids:
+                if cls_pkg := class_package_mapping.get(_type.uuid):
+                    if cls_pkg != current_pkg:
+                        pkg_dependencies.add(cls_pkg)
+                        build_in = cls_pkg in self.build_ins
+                        if not build_in:
+                            cls_pkg += self.pkg_postfix
+                        pkg = f"{cls_pkg}/"
+                else:
+                    logger.error(
+                        "Class %s was referenced in %s, but not found",
+                        _type.name,
+                        cls.name,
+                    )
+
+            type_name = pkg + (
+                _type.name if build_in else self.make_camel_case(_type.name)
+            )
+        else:
+            logger.error("Unknown type for property %s of class %s")
```

Comment:
Missing format args.

I think this should be it:

```suggestion
            logger.error(
                "Unknown type for property %r of class %s",
                type(_type).__name__,
                cls.name,
            )
```